src/core/create_manifest_data.ts

```diff
@@ -9,14 +10,28 @@ export default function create_manifest_data(cwd: string): ManifestData {
 		throw new Error(`As of Sapper 0.21, the routes/ directory should become src/routes/`);
 	}
 
+	function has_preload(file: string) {
+		const source = fs.readFileSync(path.join(cwd, file), 'utf-8');
+
+		if (/preload/.test(source)) {
+			try {
+				const { stats } = svelte.compile(source, { generate: false });
```

Comment:
I think we should use an empty warning handler here to suppress the display of any warnings while we're determining whether there's a preload. They'll be getting displayed during the main compilation anyway.

Reply:
good catch — added locally